phpstan.neon

```diff
@@ -1,7 +1,11 @@
 parameters:
-	level: 9
-	paths:
-	    - src
-	    - tests
-	bootstrapFiles:
-	    - vendor/bin/.phpunit/phpunit-9.5-0/vendor/autoload.php
+    level: 9
+    paths:
+        - src
+        - tests
+    bootstrapFiles:
+        - vendor/bin/.phpunit/phpunit-9.5-0/vendor/autoload.php
+    ignoreErrors:
+        - message: '#^Class AymDev\\MessengerAzureBundle\\Messenger\\Transport\\AzureTransportFactory implements generic interface Symfony\\Component\\Messenger\\Transport\\TransportFactoryInterface but does not specify its types\: TTransport$#'
+          path: src/Messenger/Transport/AzureTransportFactory.php
+          reportUnmatched: false
```

Comment:
With Symfony Messenger 6.4 there is new `@template-covariant` on `TransportFactoryInterface` (https://github.com/symfony/messenger/blob/6.4/Transport/TransportFactoryInterface.php#L23) which changes the return type of `createTransport` for PHPStan. Unfortunately with different PHP versions, different Messenger version is installed (Messenger 6.4 requires PHP 8.1). For now I don't know any better solution for this than to ignore the error :disappointed:

Reply:
Noted ! The package should drop unsupported PHP/Symfony versions to resolve this, so no worries :slightly_smiling_face: